### PR TITLE
add lsst_build_{repo,branch,ensure} params to lsststack::lsstsw

### PR DESCRIPTION
**Pull request by jhoblitt**
This is to enable management of the lsst_build checkout, primarily for testing,
without requiring modification to the lsstsw deploy script.

### CODE REVIEW

File: manifests/lsstsw.pp [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  $manage_user     = true,
-  $manage_group    = true,
-  $lsstsw_repo     = 'https://github.com/lsst/lsstsw.git',
-  $lsstsw_branch   = 'master',
-  $lsstsw_ensure   = 'present',
-  $buildbot_repo   = 'https://github.com/lsst-sqre/buildbot-scripts.git',
-  $buildbot_branch = 'master',
-  $buildbot_ensure = 'present',
-  $debug           = false,
+  $user              = $title,
```
Comment (ktlim) [inline manifests/lsstsw.pp]: This is why we usually discourage aligning assignments (or other things) across multiple unrelated lines.  It makes it difficult to determine what actually changed.

**Comment by ktlim** [comment]
Otherwise generally looks pretty straightforward (although it's branching the dependencies/refreshing).  Test cases also look reasonable.

File: manifests/lsstsw.pp (hunk 1 above)
Comment (jhoblitt) [inline manifests/lsstsw.pp]: I have mixed feelings about but it's idiomatic puppet.  The linter doesn't warn on this case (yet) but it will complain about unaligned hash rockets.

**Comment by jhoblitt** [comment]
I'm not sure I understand the comment about dependencies/refreshing.  Are you referring to some of the resource parameters not being checked in the unit tests?

**Comment by ktlim** [comment]
I was just contrasting the word "straightforward" with the new dependency/refresh graph implicit in the puppet source, which is no longer straight-line.  Nothing to worry about.

**Comment by jhoblitt** [comment]
Gotcha.  Thanks.
